shell/platform/linux/fl_engine.cc

```diff
@@ -573,6 +601,21 @@ gboolean fl_engine_start(FlEngine* self, GError** error) {
     g_warning("Failed to enable accessibility features on Flutter engine");
   }
 
+  std::optional<double> refresh_rate = get_refresh_rate();
```

Comment:
The refresh_rate should come from the renderer via fl_renderer_get_refresh_rate - it may differ between renderers and a headless engine has no refresh rate. This can then be added to `fl_renderer_gdk.cc`. In that you can call `gdk_display_get_monitor_at_window` which will ensure the appropriate refresh rate is used. This can also be tested by modifying `testing/mock_renderer.cc`.

Reply:
Done.